python/shotgun_data/shotgun_data_retriever.py

```diff
@@ -653,7 +813,34 @@ def _get_attachment_path(attachment_entity, bundle):
     @staticmethod
     def _get_thumbnail_path(url, bundle, directory_only=False):
         """
-        Returns the location on disk suitable for a thumbnail given its url.
+        Returns the location on disk suitable for a thumbnail given its url and
+        whether a cached file for the specified ``url`` already exists. Two cases
+        are handled:
+
+        Case A: ``directory_only`` is set to False and the ``url`` cache file does not exist:
```

Comment:
this is unrelated to the CR but what's this directory_only business? It doesn't seem safe - as far as i can tell from the code, it's stripping out a bunch of the hashing that is supposed to guarantee that the cache is unique. Can you take a quick look? It looks like it's used for the attachments download - @thebeeland, what's the reason it needs to be cached differently? Could it be done without dropping parts of the hash? 

As far as i can tell, it looks like as if all attachment for example were named foo.png there would be a high risk of hash collisions if we only use the first 4 chars of the md5. 

If the attachment filename is important, perhaps we can append it and make it part of the filename or something like that?